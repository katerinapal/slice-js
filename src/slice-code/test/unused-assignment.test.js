import {runAllCombosTests, comboOfItems} from './helpers/utils'

// test.only(
//   'unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":1}, {"index":0}, {"index":2})',
//   require('./helpers/utils').snapSlice(require.resolve('./fixtures/unused-assignment'), ({unusedAssignment}) => {
//     return [
//       unusedAssignment({index: 0}, {index: 1}, {index: 2}),
//       unusedAssignment({index: 1}, {index: 0}, {index: 2}),
//     ]
//   })
// )

runAllCombosTests({
  filename: require.resolve('./fixtures/unused-assignment'),
  methods: [
    {
      methodName: 'unusedAssignment',
      possibleArguments: comboOfItems([{index: 0}, {index: 1}, {index: 2}]),
    },
  ],
})
