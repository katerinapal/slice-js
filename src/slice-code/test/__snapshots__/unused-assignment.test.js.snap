exports[`test dependencies([false], [false]) && dependencies([false], [true]) && dependencies([true], [false]) && dependencies([true], [true]) 1`] = `
"export { dependencies };

function dependencies(foo, bar) {
  const [firstFoo] = foo;
  const [firstBar] = bar;

  if (firstFoo === firstBar) {
    return \'same\';
  } else {
    return \'different\';
  }
}"
`;

exports[`test dependencies([false], [false]) && dependencies([false], [true]) && dependencies([true], [false]) 1`] = `
"export { dependencies };

function dependencies(foo, bar) {
  const [firstFoo] = foo;
  const [firstBar] = bar;

  if (firstFoo === firstBar) {
    return \'same\';
  } else {
    return \'different\';
  }
}"
`;

exports[`test dependencies([false], [false]) && dependencies([false], [true]) && dependencies([true], [true]) 1`] = `
"export { dependencies };

function dependencies(foo, bar) {
  const [firstFoo] = foo;
  const [firstBar] = bar;

  if (firstFoo === firstBar) {
    return \'same\';
  } else {
    return \'different\';
  }
}"
`;

exports[`test dependencies([false], [false]) && dependencies([false], [true]) 1`] = `
"export { dependencies };

function dependencies(foo, bar) {
  const [firstFoo] = foo;
  const [firstBar] = bar;

  if (firstFoo === firstBar) {
    return \'same\';
  } else {
    return \'different\';
  }
}"
`;

exports[`test dependencies([false], [false]) && dependencies([true], [false]) && dependencies([true], [true]) 1`] = `
"export { dependencies };

function dependencies(foo, bar) {
  const [firstFoo] = foo;
  const [firstBar] = bar;

  if (firstFoo === firstBar) {
    return \'same\';
  } else {
    return \'different\';
  }
}"
`;

exports[`test dependencies([false], [false]) && dependencies([true], [false]) 1`] = `
"export { dependencies };

function dependencies(foo, bar) {
  const [firstFoo] = foo;
  const [firstBar] = bar;

  if (firstFoo === firstBar) {
    return \'same\';
  } else {
    return \'different\';
  }
}"
`;

exports[`test dependencies([false], [false]) && dependencies([true], [true]) 1`] = `
"export { dependencies };

function dependencies(foo, bar) {

  return \'same\';
}"
`;

exports[`test dependencies([false], [false]) 1`] = `
"export { dependencies };

function dependencies(foo, bar) {

  return \'same\';
}"
`;

exports[`test dependencies([false], [true]) && dependencies([true], [false]) && dependencies([true], [true]) 1`] = `
"export { dependencies };

function dependencies(foo, bar) {
  const [firstFoo] = foo;
  const [firstBar] = bar;

  if (firstFoo === firstBar) {
    return \'same\';
  } else {
    return \'different\';
  }
}"
`;

exports[`test dependencies([false], [true]) && dependencies([true], [false]) 1`] = `
"export { dependencies };

function dependencies(foo, bar) {

  return \'different\';
}"
`;

exports[`test dependencies([false], [true]) && dependencies([true], [true]) 1`] = `
"export { dependencies };

function dependencies(foo, bar) {
  const [firstFoo] = foo;
  const [firstBar] = bar;

  if (firstFoo === firstBar) {
    return \'same\';
  } else {
    return \'different\';
  }
}"
`;

exports[`test dependencies([false], [true]) 1`] = `
"export { dependencies };

function dependencies(foo, bar) {

  return \'different\';
}"
`;

exports[`test dependencies([true], [false]) && dependencies([true], [true]) 1`] = `
"export { dependencies };

function dependencies(foo, bar) {
  const [firstFoo] = foo;
  const [firstBar] = bar;

  if (firstFoo === firstBar) {
    return \'same\';
  } else {
    return \'different\';
  }
}"
`;

exports[`test dependencies([true], [false]) 1`] = `
"export { dependencies };

function dependencies(foo, bar) {

  return \'different\';
}"
`;

exports[`test dependencies([true], [true]) 1`] = `
"export { dependencies };

function dependencies(foo, bar) {

  return \'same\';
}"
`;

exports[`test sortRankedItems({"rank":1}, {"rank":1}) && sortRankedItems({"rank":2}, {"rank":1}) 1`] = `
"export { sortRankedItems };

function sortRankedItems(a, b) {
  var { rank: aRank } = a;
  var bRank = b.rank;

  if (aRank === bRank) {
    return \'same\';
  } else {
    return \'not same\';
  }
}"
`;

exports[`test sortRankedItems({"rank":1}, {"rank":1}) 1`] = `
"export { sortRankedItems };

function sortRankedItems(a, b) {

  return \'same\';
}"
`;

exports[`test sortRankedItems({"rank":2}, {"rank":1}) 1`] = `
"export { sortRankedItems };

function sortRankedItems(a, b) {

  return \'not same\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {

  const { index: bIndex } = b;
  const { index } = c;
  return bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":0}, {"index":2}, {"index":1}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {

  const { index: bIndex } = b;
  const { index } = c;
  return bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;
  const { index } = c;
  return aIndex > bIndex ? \'aIndex\' : bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {

  const { index: bIndex } = b;
  const { index } = c;
  return bIndex > index ? \'bIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":1}, {"index":2}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  return \'c.index\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":0}, {"index":2}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  return \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":2}, {"index":1}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":0}, {"index":2}, {"index":1}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  return \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":1}, {"index":2}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  return \'aIndex\';
}"
`;

exports[`test unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  return \'aIndex\';
}"
`;

exports[`test unusedAssignment({"index":1}, {"index":0}, {"index":2}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  return \'aIndex\';
}"
`;

exports[`test unusedAssignment({"index":1}, {"index":0}, {"index":2}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  return \'aIndex\';
}"
`;

exports[`test unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":0}, {"index":1}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":1}, {"index":2}, {"index":0}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  const aIndex = a.index;
  const { index: bIndex } = b;

  return aIndex > bIndex ? \'aIndex\' : \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":1}, {"index":2}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  return \'bIndex\';
}"
`;

exports[`test unusedAssignment({"index":2}, {"index":0}, {"index":1}) && unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  return \'aIndex\';
}"
`;

exports[`test unusedAssignment({"index":2}, {"index":0}, {"index":1}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  return \'aIndex\';
}"
`;

exports[`test unusedAssignment({"index":2}, {"index":1}, {"index":0}) 1`] = `
"export { unusedAssignment };

function unusedAssignment(a, b, c) {
  return \'aIndex\';
}"
`;
